engine/hud/InputField.hpp

```diff
@@ -0,0 +1,107 @@
+#ifndef INPUTFIELD_HPP
+#define INPUTFIELD_HPP
+
+#include "ClickableHudElement.hpp"
+#include "TextHudElement.hpp"
+
+class InputField: public ClickableHudElement, public TextHudElement {
+public:
+    /**
+     * Constructor for TextButton.
+     *
+     * Calls base class consructor and stores text object.
+     *
+     * @param text          Text object that is rendered in HUD element.
+     * @param xPos          The position of the hud element in px.
+     * @param yPos          The position of the hud element in px.
+     * @param xDim          The dimensions of the bounding box of the element in px.
+     * @param yDim          The dimensions of the bounding box of the element in px.
+     * @param onClickFn     The static function that is called when the element
+     *                      is left-clicked by the mouse.
+     * @param xAlignFlag    Determines the alignment of the bounding box.
+     * @param yAlignFlag    DetermineS the alignment of the bounding box.
+     */
+    InputField(std::shared_ptr<Text> text, int xPos, int yPos, int xDim,
+        int yDim, void (*onClickFn)()=nullptr, AlignFlag xAlignFlag=ALIGN_LEFT,
```

Comment:
Is onClickFn needed? Not expecting any clicking logic other than activating the text field